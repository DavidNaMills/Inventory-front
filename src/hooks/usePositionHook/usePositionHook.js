import react, { useState, useLayoutEffect } from 'react';

const usePositionRef = () => {
    const [posRef, setPos] = useState(0);


    const handleScroll=()=>{
        setPos(window.scrollY);
    }

    useLayoutEffect(() => {

        window.addEventListener('scroll', handleScroll)


        return () => window.removeEventListener('scroll', handleScroll)
    }, [window.scrollY]);

    return {
        posRef
    }
}

export default usePositionRef;