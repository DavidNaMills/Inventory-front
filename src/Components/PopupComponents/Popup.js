import React from 'react';
import classes from './Popup.module.scss';
import usePositionHook from '../../hooks/usePositionHook/usePositionHook';

const style = (pos) => ({
    position: 'absolute',
    top: `${pos}px`,
    left: '0px',
    width: '100%',
    height: '100vh',
    backgroundColor: 'rgba(232, 255, 253, 0.6)',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    zIndex: '9999'
})

const Popup = (props) => {
    const { children, isOpen = false, onClose = null } = props;
    const { posRef } = usePositionHook();
    
    return (
        isOpen ?

            <div style={style(posRef)}>
                <div>
                    {children}
                </div>
            </div>
            : null
    )
}

export default Popup;