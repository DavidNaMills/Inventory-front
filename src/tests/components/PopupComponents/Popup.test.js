import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import Popup from '../../../Components/PopupComponents/Popup';

jest.mock('../../../hooks/usePositionHook/usePositionHook');
import usePositionHook from '../../../hooks/usePositionHook/usePositionHook';

const tPos = 789;

const mockStyle = {
    position: 'absolute',
    top: `${tPos}px`,
    left: '0px',
    width: '100%',
    height: '100%',
    backgroundColor: 'rgba(232, 255, 253, 0.6)',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    zIndex: '9999'
}

const TempChild = () =>(<p>I am child in time</p>)


describe('<Popup/> container test suite', ()=>{
    let wrapper;

    beforeEach(()=>{
        usePositionHook.mockImplementation(()=>({
            posRef: tPos
        }));
    
        wrapper = mount(<Popup isOpen={true}><TempChild/></Popup>);
    })


    it('matches snapshot', ()=>{
        const comp = renderer.create(<Popup isOpen={true}><TempChild/></Popup>);
        const app = comp.toJSON();
        expect(app).toMatchSnapshot();
    });
    
    it('exists', ()=>{
        expect(wrapper.exists).toBeTruthy();
    });
    
    it('returns null if isOpen === false', ()=>{
        const tWrapper = mount(<Popup isOpen={false}><TempChild/></Popup>);
        expect(tWrapper.find('div').length).toBe(0);
    });
    
    it('renders 2 <div> elements, first div with inline style equal to mockStyle when isOpen === true', ()=>{
        const div = wrapper.find('div');
        expect(div.length).toBe(2);
        expect(div.at(0).props().style).toEqual(mockStyle);
    });
    
    it('renders the child component when isOpen === true', ()=>{
        const div = wrapper.find('div');
        expect(div.find(TempChild).length).toBe(1);
    });
});