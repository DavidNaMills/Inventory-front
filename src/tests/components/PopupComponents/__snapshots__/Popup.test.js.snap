// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popup/> container test suite matches snapshot 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(232, 255, 253, 0.6)",
      "display": "flex",
      "height": "100%",
      "justifyContent": "center",
      "left": "0px",
      "position": "absolute",
      "top": "789px",
      "width": "100%",
      "zIndex": "9999",
    }
  }
>
  <div>
    <p>
      I am child in time
    </p>
  </div>
</div>
`;
